library/src-bootstrapped/scala/runtime/quoted/Matcher.scala

```diff
@@ -0,0 +1,276 @@
+package scala.runtime.quoted
+
+import scala.annotation.internal.sharable
+
+import scala.quoted._
+import scala.tasty._
+
+object Matcher {
+
+  private final val debug = false
+
+  /**
+   *
+   * @param scrutineeExpr
+   * @param patternExpr
+   * @param reflection
+   * @return None if it did not match, Some(tup) if it matched where tup contains Expr[_], Type[_] or Binding[_]
+   */
+  def unapply[Tup <: Tuple](scrutineeExpr: Expr[_])(implicit patternExpr: Expr[_], reflection: Reflection): Option[Tup] = {
+    import reflection._
+
+    def treeMatches(scrutinee: Tree, pattern: Tree)(implicit env: Set[(Symbol, Symbol)]): Option[Tuple] = {
```

Comment:
Document what gets returned (also for the following methods)

Reply:
I added the documantation